vault/resource_identity_oidc_client.go

```diff
@@ -0,0 +1,156 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+const identityOIDCClientPathPrefix = "identity/oidc/client"
+
+func identityOIDCClientResource() *schema.Resource {
+	return &schema.Resource{
+		Create: identityOIDCClientCreateUpdate,
+		Update: identityOIDCClientCreateUpdate,
+		Read:   identityOIDCClientRead,
+		Delete: identityOIDCClientDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Description: "The name of the client.",
+				Required:    true,
+			},
+			"key": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Description: "A reference to a named key resource in Vault. This cannot be modified after creation.",
+				Required:    true,
```

Comment:
The key is optional in Vault 1.10. The "default" key will be used if no key is supplied.

Reply:
@austingebauer wanted to confirm — then is this field mutable and can it be reset? In that case we will not force the creation of a new resource if the named key changes.

Reply:
That makes sense. Thanks for confirming! 👍🏼

Reply:
Fixed in 1e9f8a3